### PR TITLE
Enable locals,patvars,privates unused Scalac checks [databricks]

### LINKED ISSUE #8249: Add more `unused` scalac warning flags to prevent reintroducing unused variables
Currently we only fail the build for unused imports. Xlint:unused is too broad but we can cherry-pick locals privates etc

_Originally posted by @jlowe in https://github.com/NVIDIA/spark-rapids/pull/8245_

### PR DESCRIPTION
**Pull request by gerashegalov**
Closes #8249

- Add locals,patvars,privates to Scalac unused warnings
- Fix resulting build errors

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/shims/ParquetSchemaClipShims.scala [hunk 1]
```diff
@@ -94,9 +94,6 @@ object ParquetSchemaClipShims {
     val typeAnnotation = parquetType.getLogicalTypeAnnotation
     val typeName = parquetType.getPrimitiveTypeName
 
-    def typeString =
-      if (typeAnnotation == null) s"$typeName" else s"$typeName ($typeAnnotation)"
-
     def typeNotImplemented() =
       TrampolineUtil.throwAnalysisException(s"Parquet type not yet supported: $parquetType")
```
Comment (jlowe) [inline sql-plugin/src/main/spark330/scala/com/nvidia/spark/rapids/shims/ParquetSchemaClipShims.scala:98]: Looking at the corresponding [Spark code](https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaConverter.scala#L202-L203), I believe this is suppose to be leveraging `typeString`, e.g.:
`Parquet type not yet supported: ${typeString}`

File: sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/ParquetSchemaClipShims.scala [hunk 2]
```diff
@@ -89,9 +89,6 @@ object ParquetSchemaClipShims {
     val typeAnnotation = parquetType.getLogicalTypeAnnotation
     val typeName = parquetType.getPrimitiveTypeName
 
-    def typeString =
-      if (typeAnnotation == null) s"$typeName" else s"$typeName ($typeAnnotation)"
-
     def typeNotImplemented() =
       TrampolineUtil.throwAnalysisException(s"Parquet type not yet supported: $parquetType")
```
Comment (jlowe) [inline sql-plugin/src/main/spark340/scala/com/nvidia/spark/rapids/shims/ParquetSchemaClipShims.scala:93]: Same typeString should be used comment here.

**Comment by gerashegalov** [comment]
build

**Comment by pxLi** [comment]
CI failed https://github.com/NVIDIA/spark-rapids/issues/8235

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Review by jlowe — APPROVED** [review]

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Pull request merged**